docs/docs/scripting/worldgen.md

```diff
@@ -110,6 +110,89 @@ SPVec4 spHeightGet(SPVec4 previousHeight, //if spReplacesPreviousHeight returns
 
 It's currently unknown how biome and climate assignment works, the vanilla implementation can be found in the [splugins](https://github.com/Majic-Jungle/splugins/blob/main/SPVanilla/src/SPBiome.c) repository.
 
+Objects like trees and rocks are also placed in this stage of generation.
+
+### Objects
+
+Objects that are common to multiple methods during biome generation.
+
+#### SPBiomeThreadState
+
+Allows the biome generator to get indices for various objects generated on the Lua side. 
+
+```C
+struct SPBiomeThreadState {
+	void* terrainGenerator; //private
+	void* gom; //private
+
+    // The following methods can be called to get the index from the lua typeMap for various types of objects.
+	uint16_t (*getBiomeTag)(struct SPBiomeThreadState*,char*); // Get the index for the given biome tag, eg "hot". 
+	uint32_t (*getTerrainBaseTypeIndex)(struct SPBiomeThreadState*,char*); // Get the index for the given terrain base type, eg "riverSand"
+	uint32_t (*getTerrainVariation)(struct SPBiomeThreadState*,char*); // Get the index for the given terrain variation, eg "grassSnow"
+	uint32_t (*getTerrainModification)(struct SPBiomeThreadState*,char*); // Get the index for the given terrain modification, eg "preventGrassAndSnow"
+	uint32_t (*getGameObjectTypeIndex)(struct SPBiomeThreadState*,char*); // Get the index for the given object type, eg "appleTree". Note that this method can be null and has to be checked for existence
+
+	uint32_t (*getMaterialTypeIndex)(struct SPBiomeThreadState*,char*); 
+	uint32_t (*getDecalGroupTypeIndex)(struct SPBiomeThreadState*,char*);
+
+	SPSurfaceTypeDefault (*getSurfaceDefaultsForBaseType)(struct SPBiomeThreadState*,uint32_t);
+	SPSurfaceTypeDefault (*getSurfaceDefaultsForVariationType)(struct SPBiomeThreadState*,uint32_t);
+
+	uint32_t (*getSurfaceBaseTypeForFillObjectType)(struct SPBiomeThreadState*,uint32_t);
+
+	SPRand* spRand;
+	SPNoise* spNoise1;
+	SPNoise* spNoise2;
+}
+```
+
+### Methods
+
+The following methods can be overridden to implement custom biome generation. 
+
+#### spBiomeInit
+
+- `threadState: SPBiomeThreadState*` 
+
+This method has no return value, the method is used to initialize static variables with indices of the relevant object.  
+
+#### spBiomeGetTagsForPoint
+
+It's currently unknown what this method exactly does. You can help by checking out the vanilla code!
```

Comment:
It might be best to have a mention of the 'Edit Page' button at the bottom or something similar to the note in the CONTRIBUTING.md